test/bulk-update/document-manager/document-manager.test.js

```diff
@@ -0,0 +1,108 @@
+import { expect } from '@esm-bundle/chai';
+import fs from 'fs';
+import { loadDocument, saveDocument, entryToPath } from '../../../bulk-update/document-manager/document-manager.js';
+import BaseReporter from '../../../bulk-update/reporter/reporter.js';
+
+const { pathname } = new URL('.', import.meta.url);
+
+const config = {
+  mdDir: `${pathname}mock/`,
+  siteUrl: 'https://main--bacom--adobecom.hlx.live',
+  reporter: new BaseReporter(),
+  cacheTimeMs: 0,
+  outputDir: `${pathname}output/`,
+};
+
+const mdast = {
+  type: 'root',
+  children: [
+    {
+      type: 'paragraph',
+      children: [
+        {
+          type: 'text',
+          value: 'Test file',
+        },
+      ],
+    },
+  ],
+};
+
+describe('DocumentManager', () => {
+  beforeEach(() => {
+    config.reporter = new BaseReporter();
+  });
+
+  describe('entryToPath', () => {
+    const tests = [
+      ['/', '/index'],
+      ['/test-file', '/test-file'],
+      ['/test-file/', '/test-file/index'],
+    ];
+
+    tests.forEach(([entry, path]) => {
+      it(`converts ${entry} to ${path}`, () => {
+        expect(entryToPath(entry)).to.equal(path);
+      });
+    });
+  });
+
+  describe('loadDocument', () => {
+    it('loads a local file', async () => {
+      const entry = 'test-file';
+      config.cacheTimeMs = -1;
+
+      const document = await loadDocument(entry, config);
+      expect(document).to.deep.equal({
+        entry,
+        path: 'test-file',
+        markdown: 'Test file\n',
+        mdast,
+        markdownFile: `${pathname}mock/test-file.md`,
+        url: 'https://main--bacom--adobecom.hlx.live/test-file',
+      });
+    });
+
+    it('loads a draft file', async () => {
+      const entry = '/';
+      config.mdDir = null;
+      config.siteUrl = 'https://main--bacom--adobecom.hlx.page';
+
+      const document = await loadDocument(entry, config);
+      expect(document.url).to.equal('https://main--bacom--adobecom.hlx.page/index');
+      expect(document.markdown).to.not.be.empty;
+      expect(document.mdast).to.not.be.empty;
+    });
+
+    it('loads a live file', async () => {
+      const entry = '/';
+      config.mdDir = null;
+      config.siteUrl = 'https://main--bacom--adobecom.hlx.live';
+
+      const document = await loadDocument(entry, config);
```

Comment:
This makes actual network requests yes?

Reply:
Yes, this makes actual network requests. I tried mocking adobe fetch but ran into issues.